src/include/planner/plannodes/nested_loop_join_plan_node.h

```diff
@@ -44,8 +75,12 @@ class NestedLoopJoinPlanNode : public AbstractJoinPlanNode {
    */
   NestedLoopJoinPlanNode(std::vector<std::unique_ptr<AbstractPlanNode>> &&children,
                          std::unique_ptr<OutputSchema> output_schema, LogicalJoinType join_type,
-                         common::ManagedPointer<parser::AbstractExpression> predicate)
-      : AbstractJoinPlanNode(std::move(children), std::move(output_schema), join_type, predicate) {}
+                         common::ManagedPointer<parser::AbstractExpression> predicate,
+                         std::vector<common::ManagedPointer<parser::AbstractExpression>> left_keys,
+                         std::vector<common::ManagedPointer<parser::AbstractExpression>> right_keys)
```

Comment:
What are left and right keys for a nested loop join? Why do you need anything more than the predicate?

Also these parameters need a doxygen comment

Reply:
The left and right keys are used to specify the join on conditions (which is a subtree of the predicate). Arguably, the exec engine could derive it but I think it would be simpler/cleaner if that information was in the plan node.